game/yodesk.ksy

```diff
@@ -0,0 +1,414 @@
+             '"STUP"': setup_image
+             '"CHAR"': characters
+             '"CAUX"': character_auxiliaries
+             '"CHWP"': character_weapons
+             '"PUZ2"': puzzles
+             '"SNDS"': sounds
+             '"TILE"': tiles
+             '"TNAM"': tile_names
+             '"ZONE"': zones
+             '"ENDF"': endf
+             _: unknown_case_error
+  unknown_case_error:
+    seq:
+      - id: message
+        contents: "Value did not match any expected type"
+  version:
+    seq:
+      - id: version
+        type: u4
+  setup_image:
+    seq:
+      - id: size
+        type: u4
+      - id: pixels
+        size: size
+  sounds:
+    seq:
+      - id: size
+        type: u4
+      - id: count
+        type: s2
+      - id: sounds
+        type: prefixed_strz
+        repeat: expr
+        repeat-expr: -count
+  tile_names:
+    seq:
+      - id: size
+        type: u4
+      - id: names
+        type: tile_name
+        repeat: until
+        repeat-until:  _.tile_id == -1
+  tile_name:
+    seq:
+      - id: tile_id
+        type: s2
+      - id: name
+        type: str
+        size: 0x18
+        if: tile_id != -1
+  tiles:
+    seq:
+      - id: size
+        type: u4
+      - id: tiles
+        type: tile
+        repeat: expr
+        repeat-expr: size / ( 32 * 32 + 4)
+  tile:
+      seq:
+        - id: attributes
+          type: u4
+        - id: pixels
+          size: 32 * 32
+
+  action:
+    seq:
+      - id: marker
+        contents: "IACT"
+      - id: size
+        type: u4
+      - id: condition_count
+        type: u2
+      - id: conditions
+        type: action_item
+        repeat: expr
+        repeat-expr: condition_count
+      - id: instruction_count
+        type: u2
+      - id: instructions
+        type: action_item
+        repeat: expr
+        repeat-expr: instruction_count
+  action_item:
+    seq:
+      - id: opcode
+        type: u2
+      - id: arguments
+        type: s2
+        repeat: expr
+        repeat-expr: 5
+      - id: text_length
+        type: u2
+      - id: text
+        type: str
+        size: text_length
+  monster:
+    seq:
+      - id: character
+        type: u2
+      - id: x
+        type: u2
+      - id: y
+        type: u2
+      - id: loot
+        type: s2
+      - id: drops_loot
+        type: s4
+      - id: waypoints
+        type: waypoint
+        repeat: expr
+        repeat-expr: 4
+  zone:
+    seq:
+      - id: planet
+        type: u2
+      - id: size
+        type: u4
+      - id: index
+        type: u2
+      - id: marker
+        contents: "IZON"
+      - id: size2
+        type: u4
+      - id: width
+        type: u2
+      - id: height
+        type: u2
+      - id: type
+        type: u4
+      - id: shared_counter
+        type: u2
+        # contents: 0xFF FF
+      - id: planet_again
+        type: u2
+      - id: tile_ids
+        type: u2
+        repeat: expr
+        repeat-expr: 3 * width * height
```

Comment:
Does 3 mean count of channels? If so, what is the order?

Reply:
I added a `doc:` comment to the field, explaining that each map has 3 tile layers. Hope that clears things up.

Reply:
It may make sense to encode it in KS.

Reply:
Can you take a look at cc51e05a53dd16ebcfbeff193aafc1436ad4d789 and let me know if that is what you had in mind?

Reply:
Reviewed.